File: translate.py
import os
import re

replacements = {
    r">Nome<": ">Name<",
    r">Fabricante<": ">Vendor<",
    r">Credenciais<": ">Credentials<",
    r">Rotinas<": ">Routines<",
    r">Configurações<": ">Settings<",
    r">Sistema<": ">System<",
    r">Usuários<": ">Users<",
    r">Usuário<": ">User<",
    r">Senha<": ">Password<",
    r">Salvar<": ">Save<",
    r">Sair<": ">Logout<",
    r">Voltar<": ">Back<",
    r">Editar<": ">Edit<",
    r">Excluir<": ">Delete<",
    r">Novo Node<": ">New Node<",
    r">Nova Credencial<": ">New Credential<",
    r">Novo Usuário<": ">New User<",
    r">Nova Rotina<": ">New Routine<",
    r">Ativo<": ">Active<",
    r">Inativo<": ">Inactive<",
    r">Sucesso<": ">Success<",
    r">Falha<": ">Failed<",
    r">Carregando...<": ">Loading...<",
    r">Nenhum node em estado de falha!<": ">No failed nodes!<",
    r">Nenhum backup realizado ainda.<": ">No backups yet.<",
    r">Nenhuma atividade registrada.<": ">No recent activity.<",
    r">Próximos Backups<": ">Upcoming Backups<",
    r">Últimos Backups<": ">Recent Backups<",
    r">Atividade Recente<": ">Recent Activity<",
    r">Nodes em Falha<": ">Failed Nodes<",
    r">Total de Nodes<": ">Total Nodes<",
    r">Nodes Ativos<": ">Active Nodes<",
    r">Backups Realizados<": ">Total Backups<",
    r">Falhas<": ">Failures<",
    r">Detalhes do Node<": ">Node Details<",
    r">Frequência<": ">Frequency<",
    r">Horário<": ">Time<",
    r">Todos<": ">All<",
    r">Exportar<": ">Export<",
    r">Meu Perfil<": ">My Profile<",
    r">Logs do Sistema<": ">System Logs<",
    r"value=\"Salvar\"": "value=\"Save\"",
    r"value=\"Entrar\"": "value=\"Sign In\"",
    r"value=\"Criar\"": "value=\"Create\"",
    r"placeholder=\"Buscar...\"": "placeholder=\"Search...\"",
    r"placeholder=\"Senha\"": "placeholder=\"Password\"",
    r"placeholder=\"Usuário\"": "placeholder=\"Username\"",
    r">Entrar<": ">Sign In<",
    r">Criar Conta<": ">Create Account<",
    r">Confirmar Exclusão<": ">Confirm Deletion<",
    r">Tem certeza que deseja excluir<": ">Are you sure you want to delete<",
    r">Esta ação não pode ser desfeita.<": ">This action cannot be undone.<",
    r">Cancelar<": ">Cancel<",
    r">Sim, Excluir<": ">Yes, Delete<",
}

def translate_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
        
    original = content
    for pt, en in replacements.items():
        content = re.sub(pt, en, content)
        
    if content != original:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Translated: {filepath}")

for root, _, files in os.walk('c:/mimic_backup/templates'):
    for file in files:
        if file.endswith('.html'):
            translate_file(os.path.join(root, file))

File: translate.go
package main

import (
	"fmt"
	"io/fs"
	"os"
	"path/filepath"
	"strings"
)

var replacements = map[string]string{
	">Nome<": ">Name<",
	">Fabricante<": ">Vendor<",
	">Credenciais<": ">Credentials<",
	">Rotinas<": ">Routines<",
	">Configurações<": ">Settings<",
	">Sistema<": ">System<",
	">Usuários<": ">Users<",
	">Usuário<": ">User<",
	">Senha<": ">Password<",
	">Salvar<": ">Save<",
	">Sair<": ">Logout<",
	">Voltar<": ">Back<",
	">Editar<": ">Edit<",
	">Excluir<": ">Delete<",
	">Novo Node<": ">New Node<",
	">Nova Credencial<": ">New Credential<",
	">Novo Usuário<": ">New User<",
	">Nova Rotina<": ">New Routine<",
	">Ativo<": ">Active<",
	">Inativo<": ">Inactive<",
	">Sucesso<": ">Success<",
	">Falha<": ">Failed<",
	">Carregando...<": ">Loading...<",
	">Nenhum node em estado de falha!<": ">No failed nodes!<",
	">Nenhum backup realizado ainda.<": ">No backups yet.<",
	">Nenhuma atividade registrada.<": ">No recent activity.<",
	">Próximos Backups<": ">Upcoming Backups<",
	">Últimos Backups<": ">Recent Backups<",
	">Atividade Recente<": ">Recent Activity<",
	">Nodes em Falha<": ">Failed Nodes<",
	">Total de Nodes<": ">Total Nodes<",
	">Nodes Ativos<": ">Active Nodes<",
	">Backups Realizados<": ">Total Backups<",
	">Falhas<": ">Failures<",
	">Detalhes do Node<": ">Node Details<",
	">Frequência<": ">Frequency<",
	">Horário<": ">Time<",
	">Todos<": ">All<",
	">Exportar<": ">Export<",
	">Meu Perfil<": ">My Profile<",
	">Logs do Sistema<": ">System Logs<",
	"value=\"Salvar\"": "value=\"Save\"",
	"value=\"Entrar\"": "value=\"Sign In\"",
	"value=\"Criar\"": "value=\"Create\"",
	"placeholder=\"Buscar...\"": "placeholder=\"Search...\"",
	"placeholder=\"Senha\"": "placeholder=\"Password\"",
	"placeholder=\"Usuário\"": "placeholder=\"Username\"",
	">Entrar<": ">Sign In<",
	">Criar Conta<": ">Create Account<",
	">Confirmar Exclusão<": ">Confirm Deletion<",
	">Tem certeza que deseja excluir<": ">Are you sure you want to delete<",
	">Esta ação não pode ser desfeita.<": ">This action cannot be undone.<",
	">Cancelar<": ">Cancel<",
	">Sim, Excluir<": ">Yes, Delete<",
    // Base/HTMX specific translations
    "Sair do sistema": "Logout",
    "Carregando": "Loading",
}

func main() {
	err := filepath.WalkDir("templates", func(path string, d fs.DirEntry, err error) error {
		if err != nil {
			return err
		}
		if d.IsDir() {
			return nil
		}
		if !strings.HasSuffix(path, ".html") {
			return nil
		}

		b, err := os.ReadFile(path)
		if err != nil {
			return err
		}

		content := string(b)
		original := content

		for pt, en := range replacements {
			content = strings.ReplaceAll(content, pt, en)
		}

		if content != original {
			err = os.WriteFile(path, []byte(content), 0644)
			if err != nil {
				return err
			}
			fmt.Printf("Translated: %s\n", path)
		}

		return nil
	})

	if err != nil {
		fmt.Println("Error:", err)
	}
}
